packages/parallel/init.lua

```diff
@@ -99,75 +438,196 @@ function package:_init (options)
-            calculations[frame].mark,
-            ", now",
-            #typesetter.state.outputQueue,
-            ", height of material:",
-            calculations[frame].heightOfNewMaterial
-         )
       end)
+
+      -- Add balancing glue to align frame heights.
       addBalancingGlue(maxheight)
+
+      -- Handle parskip (spacing between successive frames).
+      local parskip = SILE.settings:get("document.parskip")
+
+      if not parskip.length then
+         -- Insert default parskip value if not defined.
+         addParskipToFrames(SILE.types.length("12pt plus 3pt minus 1pt"))
+      else
+         -- Use user-defined parskip value.
+         addParskipToFrames(parskip)
+      end
+   end)
+
+   self:registerCommand("smaller", function(_, content)
+      SILE.settings:temporarily(function()
+         local currentSize = SILE.settings:get("font.size")
+         SILE.settings:set("font.size", currentSize * 0.75) -- Scale down to 75%
+         SILE.settings:set("font.weight", 800)
+         SILE.process(content)
+      end)
+   end)
+
+   self:registerCommand("footnoteNumber", function(options, content)
+      local height = options.height or "0.3em" -- Default height for superscripts
+      SILE.call("raise", { height = height }, function()
+         SILE.call("smaller", {}, function()
+            SILE.process(content)
+         end)
+      end)
+   end)
+
+   -- Adapted from the `resilient.footnotes` package.
+   -- Defines `parallel_footnote:rule`, a helper command for setting a footnote rule.
+   self:registerCommand("parallel_footnote:rule", function(options, _)
+      local width = SU.cast("measurement", options.width or "20%fw") -- Default: 1/5 of the text block width.
+      local beforeskipamount = SU.cast("vglue", options.beforeskipamount or "1ex") -- Space before the rule.
+      local afterskipamount = SU.cast("vglue", options.afterskipamount or "1ex") -- Space after the rule.
+      local thickness = SU.cast("measurement", options.thickness or "0.5pt") -- Thickness of the rule.
+
+      SILE.call("noindent")
+      SILE.call("rebox", {}, function()
+         SILE.call("hrule", { width = width, height = thickness })
+      end)
+      SILE.typesetter:leaveHmode()
+      SILE.typesetter:pushExplicitVglue(afterskipamount)
+   end, "Helper command for setting a footnote rule.")
+
+   self:registerCommand("parallel_footnote:constructor", function(options, content)
+      local markerText = options.marker or "?" -- Default marker if none provided
+
+      SILE.settings:temporarily(function()
+         -- Set font footnotes, 80% of current font size
+         SILE.settings:set("font.size", SILE.settings:get("font.size") * 0.80)
+
+         -- Measure the marker width
+         local markerWidth = measureStringWidth(markerText)
+
+         -- Set hanging indentation
+         local hangIndent = SILE.types.length("14.4pt"):absolute()
+         SILE.settings:set("current.hangAfter", 1) -- Indent subsequent lines
```

Comment:
Mere remark: note that hangAfter/hangIndent is touchy (on when it resets etc.) and overkill here.
For a 1-line indent, it can most of the time be avoided:
- set the whole left skip to your amount (document.lskip)
- set the current paragraph indent to the _negative_ of that amount ;)